src/alertmanager_client.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import json
+import logging
+import urllib.parse
+from typing import List, Optional
+
+import requests
+
+logger = logging.getLogger(__name__)
+
+
+class AlertmanagerAPIClient:
+    """Alertmanager HTTP API client."""
+
+    def __init__(self, address: str = "localhost", port: int = 9093, timeout=2.0):
+        self.base_url = f"http://{address}:{port}/"
+        self.timeout = timeout
+
+    def reload(self) -> bool:
+        """Send a POST request to to hot-reload the config.
+        This reduces down-time compared to restarting the service.
+
+        Returns:
+          True if reload succeeded (returned 200 OK); False otherwise.
+        """
+        url = urllib.parse.urljoin(self.base_url, "/-/reload")
+        try:
+            response = requests.post(url, timeout=self.timeout)
```

Comment:
```suggestion
            response = urllib.request.urlopen(url, data=None, timeout=self.timeout)
```

Perhaps drop the dependency on `requests`?

Reply:
Done